### PR TITLE
Feature/add filter in explore page

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses   #3840

## Changes

* Add Filters in Explore Deep page

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [ ] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/components/FrameworkImageButton/index.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ function FrameworkImageButton(props: Props) {
         label,
         variant,
     } = props;
+    console.warn('here', props);
```
Comment (samshara) [inline app/components/FrameworkImageButton/index.tsx:36]: remove this

File: app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+type FormType = Partial<ProjectJoinFields>;
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        reason: [
+            requiredStringCondition,
+            lengthGreaterThanCondition(100),
```
Comment (samshara) [inline app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx:33]: do we need this condition ?

File: app/views/ExploreDeep/index.tsx [hunk 3]
```diff
@@ -14,9 +85,119 @@ function ExploreDeep(props: Props) {
         className,
     } = props;
 
+    const [filters, setFilters] = useState<ProjectListQueryVariables | undefined>(undefined);
+    const [page, setPage] = useState<number>(1);
+    const [pageSize, setPageSize] = useState<number>(25);
```
Comment (samshara) [inline app/views/ExploreDeep/index.tsx:90]: we could define a const from pageSize for better semantics.

File: app/views/ExploreDeep/index.tsx [hunk 4]
```diff
@@ -14,9 +85,119 @@ function ExploreDeep(props: Props) {
...
+            title: '',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: ActionCell,
+            cellRendererParams: (projectId, project) => ({
+                projectId,
+                membershipPending: project?.membershipPending,
+                isMember: !!project?.currentUserRole,
```
Comment (samshara) [inline app/views/ExploreDeep/index.tsx:138]: we could use helper functions here

File: app/views/ExploreDeep/index.tsx [hunk 5]
```diff
@@ -14,9 +85,119 @@ function ExploreDeep(props: Props) {
...
+
+    const variables = useMemo(() => ({
+        ...filters,
+        page,
+        pageSize,
+    }), [page, pageSize, filters]);
+
+    const {
+        data,
+        loading,
```
Comment (samshara) [inline app/views/ExploreDeep/index.tsx:100]: this variable is unused. we can show status or remove it.

File: app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx (hunk 2 above)
Comment (subinasr) [reply app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx:33]: @AdityaKhatri  did this. Can you please answer this question?

Comment (AdityaKhatri) [reply app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx:33]: This is required

**Pull request closed without merging**